api/configserver/save.go

```diff
@@ -50,6 +52,14 @@ type SaveConfigResponse struct {
 }
 
 func (s *Server) SaveConfig(w http.ResponseWriter, r *http.Request) {
+	s.saveConfig(w, r, true)
+}
+
+func (s *Server) SaveConfigSkipCredentials(w http.ResponseWriter, r *http.Request) {
```

Comment:
Could this be a query param instead of a separate API endpoint? And maybe have it so you opt-in to credential checking rather than doing it by default? Something like `?check_creds`.